### ISSUE TITLE
 Exception in server tick loop  java.lang.IllegalStateException: no implementation set

### ISSUE THREAD

**Issue by OverlordWeaponry** [CitizensDev/Citizens2#1264]
Post Type: Bug

Everytime this occurs, my server crashes

`[15:59:32] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.IllegalStateException: no implementation set
	at net.citizensnpcs.api.CitizensAPI.getImplementation(CitizensAPI.java:65) ~[?:?]
	at net.citizensnpcs.api.CitizensAPI.getPlugin(CitizensAPI.java:103) ~[?:?]
	at net.citizensnpcs.api.npc.AbstractNPC$1.setPersistent(AbstractNPC.java:67) ~[?:?]
	at net.citizensnpcs.npc.skin.Skin.setNPCSkinData(Skin.java:356) ~[?:?]
	at net.citizensnpcs.npc.skin.Skin.apply(Skin.java:106) ~[?:?]
	at net.citizensnpcs.npc.skin.SkinPacketTracker.updateViewer(SkinPacketTracker.java:217) ~[?:?]
	at net.citizensnpcs.nms.v1_12_R1.util.PlayerlistTrackerEntry.updatePlayer(PlayerlistTrackerEntry.java:40) ~[?:?]
	at net.minecraft.server.v1_12_R1.EntityTracker.updatePlayers(EntityTracker.java:202) ~[patched_1.12.1.jar:git-Paper-1170]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:929) ~[patched_1.12.1.jar:git-Paper-1170]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:424) ~[patched_1.12.1.jar:git-Paper-1170]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) ~[patched_1.12.1.jar:git-Paper-1170]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [patched_1.12.1.jar:git-Paper-1170]
`
Config:
`# Citizens Configuration

general:
  authlib:
    profile-url: https://sessionserver.mojang.com/session/minecraft/profile/
  debug-file: ''
  debug-mode: false
  debug-pathfinding: false
  color-scheme:
    message-highlight: <e>
    message: <a>
  translation:
    locale: ''
npc:
  chat:
    options:
      bystanders-hear-targeted-chat: true
      max-number-of-targets-to-show: 2
      multiple-targets-format: <target>|, <target>| & <target>| & others
      range: 5
      max-text-range: 500
    format:
      no-targets: '[<npc>]: <text>'
      with-target-to-bystanders: '[<npc>] -> [<target>]: <text>'
      to-target: '[<npc>] -> You: <text>'
      with-targets-to-bystanders: '[<npc>] -> [<targets>]: <text>'
  pathfinding:
    default-distance-margin: 0.5
    update-path-rate: 20
    new-finder-open-doors: false
    attack-range: 3.0625
    use-new-finder: false
  default:
    look-close:
      enabled: false
      range: 5
    pathfinding:
      range: 25.0
    random-talker: true
    realistic-looking: false
    stationary-ticks: -1
    talk-close:
      enabled: false
      range: 5
    text:
      '0': Hi, I'm <npc>!
  limits:
    default-limit: 10
    max-permission-checks: 100
    max-packet-entries: 15
    max-speed: 100
  tablist:
    disable: true
  chunks:
    always-keep-loaded: false
  skins:
    max-retries: -1
    retry-delay: 120
    rotation-update-degrees: 90.0
    use-latest: true
    view-distance: 100.0
  packets:
    update-delay: 30
  selection:
    quick-select: false
    item: '280'
    message: <b>You selected <a><npc><b>!
  player:
    remove-from-list: true
  server-ownership: false
  text:
    max-talk-cooldown: 5
    min-talk-cooldown: 10
    talk-item: '340'
  teleport-delay: -1
  controllable:
    use-boat-controls: true
  player-scoreboard-teams:
    enable: true
advanced:
  check-minecraft-version: true
economy:
  npc:
    cost: 100.0
storage:
  save-task:
    delay: 72000
  file: saves.yml
  type: yaml
subplugins:
  folder: plugins
`
Minecraft Version: 1.12.1 (Server)
Citizens Version: citizens-2.0.22-SNAPSHOT (Latest build from Dev Builds)

**Comment by mcmonkey4eva** [comment]
Can you post THROUGH PASTEBIN a full log?
That error appears to be Citizens still trying to run after something else crashed the server...

**Comment by OverlordWeaponry** [comment]
Sorry about that. Here's a pastebin of what happened.

https://pastebin.com/xNywDgGb

Also, I would like to note that before the server crashed, a copy of another NPC showed up, one that wasn't supposed to. When I touched it, I got disconnected. When I log back in, the server crashes.

Also, I'll be moving to the stable build on the Bukkit Website. I'm using a Dev build after all, and those can be shaky.
